File: Algorithms/C_Sharp/ProblemSolving/Solutions/Easy/TwoSum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProblemSolving.Solutions.Easy
{
    internal class TwoSum
    {
        public int[] Solution(int[] nums, int target)
        {
            for (int i = 0; i < nums.Length - 1; i++)
            {
                for (int j = i + 1; j < nums.Length; j++)
                {
                    if (nums[i] + nums[j] == target)
                    {
                        return new int[] { i, j };
                    }
                }
            }
            throw new InvalidOperationException("No two elements of an array can add up to the target!");
        }

        public void Run()
        {
            try
            {
                Console.Write("Enter elements of an array separating with single space: ");
                int[] numbers = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
                Console.Write("Enter target number: ");
                int targetNumber = int.Parse(Console.ReadLine());
                int[] result = Solution(numbers, targetNumber);
                Console.WriteLine($"Result: [{string.Join(",", result)}]");
            }
            catch (InvalidOperationException e)
            {
                Console.WriteLine(e.Message);
            }

        }
    }
}

File: Algorithms/C_PlusPlus/ProblemSolving/TwoSum.cpp
#include "TwoSum.h"
using namespace std;

int Run()
{
    std::vector<int> nums;
    int target;

    // Read input
    int size;
    std::cout << "Enter the number of elements in the array: ";
    std::cin >> size;

    nums.resize(size);
    std::cout << "Enter the elements of the array: ";
    for (int i = 0; i < size; i++) {
        std::cin >> nums[i];
    }

    std::cout << "Enter the target sum: ";
    std::cin >> target;

    // Find the indices of the two numbers
    std::vector<int> indices = twoSum(nums, target);

    // Output the result
    std::cout << "Indices of the two numbers that add up to the target: " << indices[0] << " and " << indices[1] << std::endl;

    return 0;
};


